web/rtc/webrtc.c

```diff
@@ -0,0 +1,740 @@
+    } link;
+} WEBRTC_DC;
+
+typedef struct webrtc_connection {
+    int pc;
+    rtcConfiguration config;
+    rtcState state;
+    rtcGatheringState gathering_state;
+
+    size_t max_message_size;
+    size_t local_max_message_size;
+    size_t remote_max_message_size;
+
+    struct {
+        SPINLOCK spinlock;
+        BUFFER *wb;
+        bool sdp;
+        bool candidates;
+    } response;
+
+    struct {
+        SPINLOCK spinlock;
+        WEBRTC_DC *head;
+    } channels;
+
+    struct {
+        struct webrtc_connection *prev;
+        struct webrtc_connection *next;
+    } link;
+} WEBRTC_CONN;
+
+#define WEBRTC_MAX_ICE_SERVERS 100
+
+static struct {
+    bool enabled;
+    char *iceServers[WEBRTC_MAX_ICE_SERVERS];
+    int iceServersCount;
+    char *proxyServer;
+    char *bindAddress;
+
+    struct {
+        SPINLOCK spinlock;
+        WEBRTC_CONN *head;
+    } unsafe;
+
+} webrtc_base = {
+#ifdef NETDATA_INTERNAL_CHECKS
+        .enabled = true,
+#else
+        .enabled = false,
+#endif
+        .iceServers = {
+                // Format:
+                // [("stun"|"turn"|"turns") (":"|"://")][username ":" password "@"]hostname[":" port]["?transport=" ("udp"|"tcp"|"tls")]
+                //
+                // Note transports TCP and TLS are only available for a TURN server with libnice as ICE backend and govern only the
+                // TURN control connection, meaning relaying is always performed over UDP.
+                //
+                // If the username or password of a URI contains reserved special characters, they must be percent-encoded.
+                // In particular, ":" must be encoded as "%3A" and "@" must by encoded as "%40".
+
+                "stun://stun.l.google.com:19302",
+                NULL, // terminator
+        },
+        .iceServersCount = 1,
+        .proxyServer = NULL, // [("http"|"socks5") (":"|"://")][username ":" password "@"]hostname["    :" port]
+        .bindAddress = NULL,
+        .unsafe = {
+                .spinlock = NETDATA_SPINLOCK_INITIALIZER,
+                .head = NULL,
+        },
+};
+
+static inline bool webrtc_dc_is_open(WEBRTC_DC *chan) {
+    return __atomic_load_n(&chan->open, __ATOMIC_RELAXED);
+}
+
+static void webrtc_config_ice_servers(void) {
+    BUFFER *wb = buffer_create(0, NULL);
+
+    int i;
+    for(i = 0; i < WEBRTC_MAX_ICE_SERVERS ;i++) {
+        if (webrtc_base.iceServers[i]) {
+            if (buffer_strlen(wb))
+                buffer_strcat(wb, " ");
+
+            internal_error(true, "WEBRTC: default ice server No %d is: '%s'", i, webrtc_base.iceServers[i]);
+            buffer_strcat(wb, webrtc_base.iceServers[i]);
+        }
+        else
+            break;
+    }
+    webrtc_base.iceServersCount = i;
+    internal_error(true, "WEBRTC: there are %d default ice servers: '%s'", webrtc_base.iceServersCount, buffer_tostring(wb));
+
+    char *servers = config_get(CONFIG_SECTION_WEBRTC, "ice servers", buffer_tostring(wb));
```

Comment:
should this be in `cloud.conf` or under `aclk` so it is clear it is cloud related?